WordPress/gutenberg
# Update Medium logo for Social Link Block

## What?
Updates the Medium logo for the Social Link Block

## Why?
Issue: https://github.com/WordPress/gutenberg/issues/57064
Medium updated its logo in 2020.

## How?
Change the svg in two places. Note that I kept the spacing noted in [Medium's brand guidelines](https://medium.design/logos-and-brand-guidelines-f1a01a733592). I'm inclined to remove at least some of that space now that I see it it place beside other logos.

## Testing Instructions
1. Add a Social Links block and add at least the Medium social link
2. Notice that it has the correct "three elipse" symbol instead of the capital "M" logo.

## Screenshots or screencast
<img width="253" alt="Screenshot 2023-12-14 at 12 22 21 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/d263bb86-d6d3-405b-83db-672d9ff22394">

<img width="290" alt="Screenshot 2023-12-14 at 12 22 41 PM" src="https://github.com/WordPress/gutenberg/assets/2152870/0a1a8174-6559-438f-9727-0e59fbd13668">
